### ISSUE TITLE
btn-check: When $enable-shadows is true, all checkboxes and radio buttons using the button toggles when checked and is focus-visible do not show a focus ring box shadow around the button toggle.

### ISSUE THREAD

**Issue by classicmike** [twbs/bootstrap#39569]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I have been using these button toggle buttons (https://getbootstrap.com/docs/5.3/forms/checks-radios/#toggle-buttons) for our checkboxes and radios. There seems to be a bug where if in your SCSS you set the ```$enable-shadows``` to true, these radio toggles when checked will not show a box shadow whenever a user is using a keyboard and focuses on the checkbox/radio for the toggle buttons like you normally see for standard buttons. i.e: on ```focus-visible``` and ```checked``` state at the same time, no focus ring box shadow is shown. These button toggles are fine on unchecked focused state.

This is an accessibility issue and will for sure fail WCAG focus-visible criteria, if kept in the current state.

This bug can be replicated on the default SCSS styles that you get out of the box from the bootstrap framework, as long as you then set the ```$enable-shadows``` to true. I've had a check and it seems like by default ```$enable-shadows``` is set to false and when it is set to false, the focus ring box shadow works perfectly on checked state. Below is are screenshots with HTML based on what was provided in BS documentation:

### Screenshots when ```$enable-shadows``` is set to false:
![Screenshot showing the input checkbox with focus:visible state and is checked](https://github.com/twbs/bootstrap/assets/1841040/b7ce97f2-a774-447e-94a6-09fa9b581af9)
![Screenshot showing correctly with a focus ring on focus:visible state and is checked](https://github.com/twbs/bootstrap/assets/1841040/853a9f06-2331-489e-888b-0987a346686b)

### Screenshots when ```$enable-shadows``` is set to true:
Note: The middle toggle button is the one that is being checked.
![Screenshot showing the input checkbox with focus:visible state and is checked](https://github.com/twbs/bootstrap/assets/1841040/1a796750-13fc-46b9-bd28-8a1692f3959a)
![Screenshot showing incorrectly with no focus ring on focus:visible state and is checked](https://github.com/twbs/bootstrap/assets/1841040/5afb96bf-8378-4452-9ffd-5fbe42201aca)

### Minimum steps to replicate
1. Install Bootstrap as a framework.
2. Setup the SCSS as per standard instructions
3. Add an additional line in the SCSS code ```$enable-shadows``` to ```true```.
4. Load the page, open up the inspector and set the checkbox/radio with the ```btn-check``` CSS class to have a ```focus-visible``` state. Make sure the checkbox is checked. This will simulate a situation when the user is using a keyboard to tab through the checkboxes/radios.
5. You should be able to see that there is no focus ring around the button toggle.

### Code demo when ```$enable-shadows``` is ```true``` and the bug shows itself:
https://stackblitz.com/edit/stackblitz-starters-3jpccy?file=index.html

### Code demo when ```$enable-shadows``` is ```false``` and the toggle checkboxes work fine:
https://stackblitz.com/edit/stackblitz-starters-uaqdud?file=index.html

### What operating system(s) are you seeing the problem on?

Windows, macOS, Android, iOS, Linux

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.3.2

**Comment by classicmike** [comment]
Thanks @louismaximepiton
